### PR TITLE
See datasets organization

### PR DESCRIPTION
**Pull request by l-vincent-l**
image

partial_fix #44

### CODE REVIEW

File: lib/transport_web/templates/user/organizations.html.eex [hunk 1]
```diff
@@ -2,7 +2,7 @@
 <h2><%= gettext "declare_organization" %></h1>
 <%= if @has_organizations do %>
 
-<div class="user-organizations">
+<div class="cards">
 <%= for organization <- @organizations do %>
     <div>
```
Comment (maukoquiroga) [inline lib/transport_web/templates/user/organizations.html.eex]: That one should be ```card``` or something like that.

File: lib/transport_web/controllers/user_controller.ex [hunk 2]
```diff
@@ -1,40 +1,49 @@
...
-    |> put_flash(:info, gettext "connection_needed")
-    |> redirect(to: "/login/explanation")
+    |> Client.me
+    |> case do
+     {:ok, response} ->
+       conn
+       |> assign(:has_organizations, Enum.empty?(response["organizations"]) == false)
+       |> assign(:organizations, response["organizations"])
+       |> render("organizations.html")
+     {:error, _} -> conn |> render("500.html")
```
Comment (maukoquiroga) [inline lib/transport_web/controllers/user_controller.ex]:
```elixir
conn
|> put_status(:internal_server_error)
|> render(ErrorView, "500.html")
```

File: lib/transport_web/controllers/user_controller.ex [hunk 3]
```diff
@@ -1,40 +1,49 @@
...
+    end
   end
 
-  # There was an error when requesting datagouvfr API
-  def organizations({:error, _}, conn) do
-    conn
-    |> render("500.html")
+  def org_datasets(conn, %{"slug" => slug}) do
```
Comment (maukoquiroga) [inline lib/transport_web/controllers/user_controller.ex]: Why ```org``` and not ```organization``` datasets, or just another controller ?

File: client/stylesheets/_mixins.scss [hunk 4]
```diff
@@ -51,3 +51,44 @@
     @content;
   }
 }
+
+@mixin cards {
```
[the remark this answers is not in the document]
Comment (maukoquiroga) [reply client/stylesheets/_mixins.scss]: Also should be split into ```cards``` or ```cards-container``` or ```user-organizations``` and ```card```. That way we avoid the ```> div```.

File: lib/transport/datagouvfr/client.ex [hunk 5]
```diff
@@ -32,13 +47,44 @@ defmodule Transport.Datagouvfr.Client do
...
+  end
+
+  @doc """
+  Call to GET /api/1/organizations/:slug/datasets/
+  You can see documentation here: http://www.data.gouv.fr/fr/apidoc/#!/organizations/list_organization_datasets
+  """
+  @spec datasets(map) :: {atom, [map]}
+  def datasets(%{:organization => slug}) do
+    case get(Path.join(["organizations", slug, "datasets"]),
              [timeout: 50_000, recv_timeout: 50_000]) do
```
Comment (maukoquiroga) [inline lib/transport/datagouvfr/client.ex]: Indentation

File: lib/transport_web/templates/user/organizations.html.eex [hunk 6]
```diff
@@ -12,7 +12,7 @@
           <%= organization["name"] %>
         </div>
         <div>
-            <a href="mailto:contact@transport.data.gouv.fr" class="button" role="link">
+            <a href=<%= user_path @conn, :org_datasets, organization["slug"] %> class="button" role="link">
```
Comment (maukoquiroga) [inline lib/transport_web/templates/user/organizations.html.eex]:
```elixir
   <%= user_path(@conn, :organization_datasets, organization.slug %>
```

?

That involves:
- use parenthesis
- change route name
- use a Struct

File: client/stylesheets/components/_user_org_datasets.scss [hunk 7]
```diff
@@ -0,0 +1,3 @@
+.userorgdatasets {
```
[the remark this answers is not in the document]
Comment (l-vincent-l) [reply client/stylesheets/components/_user_org_datasets.scss]: outdated

File: lib/transport_web/controllers/user_controller.ex [hunk 8]
```diff
@@ -1,40 +1,49 @@
...
-    |> assign(:organizations, response["organizations"])
-    |> render("organizations.html")
-  end
-
-  #I'm logged in, the first param is a user
-  def organizations(user, conn) when is_map(user) do
-    %{:apikey => user["apikey"]}
-    |> Client.me
-    |> organizations(conn)
+    |> get_session(:current_user)
```
[the remark this answers is not in the document]
Comment (l-vincent-l) [reply lib/transport_web/controllers/user_controller.ex]: I don't assign here

File: lib/transport_web/router.ex [hunk 9]
```diff
@@ -24,6 +24,7 @@ defmodule TransportWeb.Router do
 
     scope "/user" do
       get "/organizations", UserController, :organizations
+      get "/organizations/:slug/datasets/", UserController, :org_datasets
```
[the remark this answers is not in the document]
Comment (l-vincent-l) [reply lib/transport_web/router.ex]: Nop, this is a collection, the bug is on organizations/

File: lib/transport_web/templates/user/organizations.html.eex (hunk 6 above)
Comment (l-vincent-l) [reply lib/transport_web/templates/user/organizations.html.eex]: I don't want to use struct now.

File: lib/transport_web/controllers/user_controller.ex [hunk 10]
```diff
@@ -1,40 +1,49 @@
 defmodule TransportWeb.UserController do
   use TransportWeb, :controller
   alias Transport.Datagouvfr.Client
```
[the remark this answers is not in the document]
Comment (l-vincent-l) [reply lib/transport_web/controllers/user_controller.ex]: done

File: lib/transport/datagouvfr/client.ex [hunk 11]
```diff
@@ -32,13 +47,44 @@ defmodule Transport.Datagouvfr.Client do
...
+  Call to GET /api/1/organizations/:slug/
+  You can see documentation here: http://www.data.gouv.fr/fr/apidoc/#!/organizations/get_organization
   """
-  @spec me(map) :: {atom, [map]}
-  def me(%{:apikey => apikey}) do
-    case get("me",
-             [{"X-API-KEY", apikey}],
+  @spec organizations(String.t) :: {atom, [map]}
+  def organizations(slug) do
+    case get(Path.join("organizations", slug),
```
[the remark this answers is not in the document]
Comment (l-vincent-l) [reply lib/transport/datagouvfr/client.ex]: I'm not in favor of it here.

**Pull request merged**